Dependency injection/Hilt.md

```diff
@@ -0,0 +1,178 @@
+
+```groovy
+android {
+  ...
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
+}
+```
+
+### Annotations
+
+Let's start by listing the most important annotations. Later we will expand by explaining how each of them works.
+
+  * `@HiltAndroidApp` - Place above your application class to get the magic going. It generates all previously mentioned components and even some modules that you might need. This one is required for everything to work.
+  * `@AndroidEntryPoint` - This one is placed above Activities and Fragments. After doing so, the generated builder modules will be populated with your screens. This is required for each screen that you want to inject to. If you are using ViewModels in your application, this annotation will make sure that the ViewModel can be injected.
+  * `@Inject` - Placed above or beside the dependency that you want to inject.
+  * `@Module` - Marks a class as a module that provides some dependencies. Each module must also be annotated with `@InstallIn` annotation to specify it's scope.
+  * `@InstallIn` - Takes a single parameter that is a component to install the module in. Components have different scopes so make sure that your module is included only where needed
+  * `@Provides` - Place above the function in module that returns an instance of the dependency that you want to inject somewhere.
+  * `@Binds` - Also used in modules, but the function can be abstract or an interface function. Intead of creating and returning the instance, the function must return dependency type (usually an interface) and take the concrete implementation as a parameter.
+  * `@HiltViewModel` - Place above ViewModels to populate the `ViewModelComponent` enabling view models to be injected.
+
+
+### How it works
+
+In Hilt, everything happens through annotations. As I mentioned before, the first one you need to add is `@HiltAndroidApp` above your application class. This will generate all the components that you will need.
+
+Keep note of the scope of each generated component. For example, if some dependenci is used in ViewModels only, you need to install it in the `ViewModelComponent`.
+
+`@AndroidEntryPoint` is the next annotation that you will probably use the most. Placed above Activity or Fragment, it makes them viable for injection by adding them to Hilt generated builder modules. No further action required.
+
+`@HiltViewModel` does the same, just for view models. Once you add this annotation, you can feel free to `@Inject` dependencies to your view models. As long as dependencies are provided in some `@Module`.
+
+`@Module` and `@InstallIn` are the pair you will want to use with your modules. Every module consist of either `@Provides` or `@Binds` annotated functions that will provide dependencies as explained earlier.
+
+`@Inject` placed beside a dependency in the `constructor`, `field` or `method` will make sure that your dependency is injected where needed.
```

Comment:
```suggestion
`@Inject` is placed beside a dependency in the `constructor`, `field` or `method`, and it will make sure that your dependency is injected where needed.
```